[Edit] Python Strings: .startswith() (#6542)
Codecademy/docs: content/python/concepts/strings/terms/startswith/startswith.md

---
Title: '.startswith()'

> Description: 'Checks whether a string starts with a given value.'

Reviewer: ```suggestion
Description: 'Checks whether a string starts with a specified value.'
```

Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Functions'
  - 'Methods'
  - 'Python'
  - 'Strings'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

> The **`.startswith()`** method checks a value against a given string and returns `True` if the string starts with that value. Otherwise, it returns `False`. This method is useful in many scenarios, such as parsing data, filtering results, validating input, or processing text files.

Reviewer: The description should be repeated in the introduction.
```suggestion
The **`.startswith()`** method in Python checks whether a string begins with a specified value and returns `True` if it does. Otherwise, it returns `False`. This method is useful in many scenarios, such as parsing data, filtering results, validating input, or processing text files.
```

## Syntax

```pseudo

> str.startswith(value, start, end)

Reviewer: Revert back to the older syntax
```suggestion
string.startswith(value, start, end)
```

Reviewer: In the entire doc replace str with string wherever required. @Sriparno08

```

**Parameters:**

> - `str`: The input string against which to check the given value.

Reviewer: ```suggestion
- `string`: The input string against which to check the given value.
```

- `value`: A string or [tuple](https://www.codecademy.com/resources/docs/python/tuples) of strings to check for in the input string.
- `start` (Optional): The position in the input string where the search starts.
- `end` (Optional): The position in the input string where the search ends.

**Return Value:**

The `.startswith()` method returns `True` if the input string starts with the given value and `False` if it happens otherwise.

## Example 1: `.startswith()` Without `start` and `end` Parameters

The following example uses the Python `.startswith()` method without the `start` and `end` parameters to check if the input string starts with the given string:

```py
# Create a string

> str = "Coding is fun!"

Reviewer: ```suggestion
string = "Coding is fun!"
```

# Check if the input string starts with "Coding"

> res = str.startswith("Coding")

Reviewer: ```suggestion
res = string.startswith("Coding")
```

```shell
True
```

## Example 2: `.startswith()` with `start` and `end` Parameters

The following example uses `.startswith()` in Python with the `start` and `end` parameters to check if the given range of indices in the input string starts with the given string:

```py
# Create a string
str = "Python is a popular programming language."

# Check if indices 12-24 in the input string starts with "popular"
res = str.startswith("popular", 12, 24)

# Print the result
print(res)
```

Since indices 12-24 in `str` starts with "popular", the above code produces the following output:

```shell
True
```

## Codebyte Example: `.startswith()` with a Tuple of Strings

The following codebyte example uses the Python `.startswith()` method to determine if the input string starts with any of the strings in the given tuple:

```codebyte/python
# Create a string
str = "Codecademy is helpful!"

# Check if the input string starts with "Codecademy", "Google", or "Twitter"
res = str.startswith(("Codecademy", "Google", "Twitter"))

# Print the result
print(res)
```

Since `str` starts with "Codecademy", the output for the above code will be `True`.

## Frequently Asked Questions

<details>

>   <summary>How do .endswith() and .startswith() differ?</summary>

Reviewer: ```suggestion
  <summary>How do `.endswith()` and `.startswith()` differ?</summary>
```

>   <p>.startswith() checks if a given string starts with a specific value, whereas .endswith() checks if a given string ends with a specific value.</p>

Reviewer: Add backticks to code blocks

</details>

<details>
  <summary>Is .startswith() case-sensitive?</summary>
  <p>Yes, .startswith() is case-sensitive.</p>
</details>

<details>
  <summary>What happens if the prefix in .startswith() is longer than the input string?</summary>
  <p>If the prefix in .startswith() is longer than the input string, the method returns 'False'.</p>

> </details>

Reviewer: Change the questions into H3s
